Assets/Scripts/Doors/DoorAnimator.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Mirror;
+using UnityEngine;
+
+public class DoorAnimator : NetworkBehaviour
+{
+    private List<GameObject> insidePlayers = new List<GameObject>();
+    [SerializeField] private float openSpeed = 3f;
+    [SerializeField] private Transform leftPanel;
+    [SerializeField] private Transform rightPanel;
+
+    private Vector3 closedLeft;
+    private Vector3 closedRight;
+
+    [SerializeField] private Vector3 openLeft;
+    [SerializeField] private Vector3 openRight;
+    private float lerpTime;
+    [SyncVar]
+    private bool isOpening;
+
+    private bool isOpeningLocal;
+    [SyncVar]
+    private bool isClosing;
+
+    private bool isClosingLocal;
+    [SyncVar]
+    private bool isOpened;
+
+    private int playerLayer;
+    private int playersInTrigger;
+
+    [SerializeField] private AudioSource openSfx;
+    [SerializeField] private AudioSource closeSfx;
+    
+
+    void Awake()
+    {
+        closedLeft = leftPanel.localPosition;
+        closedRight = rightPanel.localPosition;
+        playerLayer = LayerMask.NameToLayer("Player");
+    }
+
+    public override void OnStartClient()
+    {
+        base.OnStartClient();
+        SyncInitialState();
+    }
+
+    void SyncInitialState()
+    {
+        if (isServer) return;
+        if (isOpened)
+        {
+            leftPanel.localPosition = openLeft;
+            rightPanel.localPosition = openRight;
+        }
+    }
+    
+    public void OnTriggerEnter(Collider other)
+    {
+        
+        if (other.gameObject.layer != playerLayer || !isServer) return;
+        
+        if (!insidePlayers.Contains(other.gameObject))
+        {
+            insidePlayers.Add(other.gameObject);
+        }
+    }
+
+    public void OnTriggerExit(Collider other)
+    {
+        if (other.gameObject.layer != playerLayer || !isServer) return;
+        
+        if (insidePlayers.Contains(other.gameObject))
+        {
+            insidePlayers.Remove(other.gameObject);
+        }
+    }
+
+    void OpenDoor()
+    {
+        if (isClosingLocal || isOpeningLocal) return;
+        
+        lerpTime = 0f;
+        StartCoroutine(OpenDoorAnim());
+        openSfx.Play();
+    }
+
+    void CloseDoor()
+    {
+        if (isClosingLocal || isOpeningLocal) return;
+        
+        lerpTime = 0f;
+        StartCoroutine(CloseDoorAnim());
+        closeSfx.Play();
+    }
+
+    IEnumerator OpenDoorAnim()
+    {
+        if(isServer) isOpening = true;
+        isOpeningLocal = true;
+        while (lerpTime < 1f)
+        {
+            lerpTime += Time.deltaTime * (openSpeed * 2f);
+
+            leftPanel.localPosition = Vector3.Lerp(closedLeft, openLeft, lerpTime);
+            rightPanel.localPosition = Vector3.Lerp(closedRight, openRight, lerpTime);
+            yield return new WaitForEndOfFrame();
```

Comment:
Is there a specific reason this is `WaitForEndOfFrame` or can it just be `null`?